apps/aecore/src/txs/aec_tx.erl

```diff
@@ -171,27 +161,25 @@ process_single(Tx, Trees, Height) ->
     Mod = tx_dispatcher:handler(Tx),
     Mod:process(Tx, Trees, Height).
 
--spec grant_fee_to_miner(list(tx()), trees(), height(), non_neg_integer()) ->
+-spec grant_fee_to_miner(list(signed_tx()), trees(), non_neg_integer(), height()) ->
                                 trees().
-grant_fee_to_miner([], Trees, _Height, 0) ->
-    lager:debug("No transactions in genesis block"),
+grant_fee_to_miner([], Trees, 0, _Height) ->
+    lager:info("Empty block -- no fee"),
     Trees;
-grant_fee_to_miner(Txs, Trees0, Height, TotalFee) ->
-    %% Consider creation of aec_accounts_service,
-    %% which will take state trees, height and list of (pubkey, operation) pairs
-    %% (valuable also during txs processing)
-    case [ AccountPubkey || #coinbase_tx{account = AccountPubkey} <- Txs] of
-      [] ->
-        lager:info("Invalid coinbase_tx transaction in block -- no fee"),
-        Trees0;
-      [MinerPubkey] ->
-        AccountsTrees0 = aec_trees:accounts(Trees0),
-
-        {ok, Account0} = aec_accounts:get(MinerPubkey, AccountsTrees0),
-        {ok, Account} = aec_accounts:earn(Account0, TotalFee, Height),
-
-        {ok, AccountsTrees} = aec_accounts:put(Account, AccountsTrees0),
-        Trees = aec_trees:set_accounts(Trees0, AccountsTrees),
-        Trees
+grant_fee_to_miner(SignedTxs, Trees0, TotalFee, Height) ->
+    CoinbaseTxs = lists:filter(fun is_coinbase/1, SignedTxs),
+    case CoinbaseTxs of
+        [] ->
+            lager:info("Invalid coinbase_tx transaction in block -- no fee"),
+            Trees0;
+        [SignedCoinbaseTx] ->
+            #coinbase_tx{account = MinerPubkey} = aec_tx_sign:data(SignedCoinbaseTx),
+            AccountsTrees0 = aec_trees:accounts(Trees0),
+
+            {ok, Account0} = aec_accounts:get(MinerPubkey, AccountsTrees0),
+            {ok, Account} = aec_accounts:earn(Account0, TotalFee, Height),
+
+            {ok, AccountsTrees} = aec_accounts:put(Account, AccountsTrees0),
+            Trees = aec_trees:set_accounts(Trees0, AccountsTrees),
```

Comment:
The assignment and next line is unnecessary, but ok for now...

Reply:
Agree, but this is just emacs reformat here.. I may revert it if you want.

Reply:
I mean the `Trees = ..., Trees` part?!

Reply:
Never change indentation of badly written code ;-)